reference/5.1/PSReadline/About/about_PSReadline.md

```diff
@@ -0,0 +1,959 @@
+```csharp
+void Delete(int start, int length)
+```
+
+Delete length characters from start.  This operation supports undo/redo.
+
+```csharp
+void Ding()
+```
+
+Perform the Ding action based on the users preference.
+
+```csharp
+void GetBufferState([ref] string input, [ref] int cursor)
+void GetBufferState([ref] Ast ast, [ref] Token[] tokens,
+  [ref] ParseError[] parseErrors, [ref] int cursor)
+```
+
+These two functions retrieve useful information about the current state of
+the input buffer.  The first is more commonly used for simple cases.  The
+second is used if your binding is doing something more advanced with the Ast.
+
+```csharp
+IEnumerable[Microsoft.PowerShell.KeyHandler]
+  GetKeyHandlers(bool includeBound, bool includeUnbound)
+
+```
+
+This function is used by Get-PSReadLineKeyHandler and probably isn't useful in
+a custom key binding.
+
+```csharp
+Microsoft.PowerShell.PSConsoleReadLineOptions GetOptions()
+```
+
+This function is used by Get-PSReadLineOption and probably isn't too useful in
+a custom key binding.
+
+```csharp
+void GetSelectionState([ref] int start, [ref] int length)
+```
+
+If there is no selection on the command line, -1 will be returned in both
+start and length. If there is a selection on the command line, the start and
+length of the selection are returned.
+
+```csharp
+void Insert(char c)
+void Insert(string s)
+```
+
+Insert a character or string at the cursor.  This operation supports undo/redo.
+
+```csharp
+string ReadLine(runspace remoteRunspace,
+  System.Management.Automation.EngineIntrinsics engineIntrinsics)
+```
+
+This is the main entry point to PSReadLine. It does not support recursion, so
+is not useful in a custom key binding.
+
+```csharp
+void RemoveKeyHandler(string[] key)
+```
+
+This function is used by Remove-PSReadLineKeyHandler and probably isn't too
+useful in a custom key binding.
+
+```csharp
+void Replace(int start, int length, string replacement)
+```
+
+Replace some of the input. This operation supports undo/redo. This is
+preferred over Delete followed by Insert because it is treated as a single
+action for undo.
+
+```csharp
+void SetCursorPosition(int cursor)
+```
+
+Move the cursor to the given offset. Cursor movement is not tracked for undo.
+
+```csharp
+void SetOptions(Microsoft.PowerShell.SetPSReadLineOption options)
+```
+
+This function is a helper method used by the cmdlet Set-PSReadLineOption, but
+might be useful to a custom key binding that wants to temporarily change a
+setting.
+
+```csharp
+bool TryGetArgAsInt(System.Object arg, [ref] int numericArg,
+  int defaultNumericArg)
+```
+
+This helper method is used for custom bindings that honor DigitArgument. A
+typical call looks like
+
+```powershell
+[int]$numericArg = 0
+[Microsoft.PowerShell.PSConsoleReadLine]::TryGetArgAsInt($arg,
+  [ref]$numericArg, 1)
+```
+
+## NOTE
+
+### POWERSHELL COMPATIBILITY
+
+PSReadLine requires PowerShell 3.0, or newer, and the console host. It
+does not work in PowerShell ISE.
```

Comment:
If we mention it doesn't work in ISE, should we mention it does work in VSCode?

Reply:
Fixed